src/GraphQLController.php

```diff
@@ -126,11 +132,17 @@ protected function queryContext(string $query, ?array $variables, string $schema
     /**
      * Note: it's expected this is called even when APQ is disabled to adhere
      *       to the negotiation protocol.
+     * @return array{query:string,parsedQuery:?\GraphQL\Language\AST\DocumentNode}
      */
-    protected function handleAutomaticPersistQueries(string $schemaName, OperationParams $operation): string
+    protected function handleAutomaticPersistQueries(string $schemaName, OperationParams $operation): array
     {
         $query = $operation->query;
 
+        $datum = [
+            'query' => $query,
```

Comment:
see [comment](https://github.com/rebing/graphql-laravel/pull/740#issuecomment-818117141), maybe it can clean something

Reply:
Good hint, will investigate!

Reply:
>  I have found that the query seems be accessed via ->definitions->offsetGet(0)->loc->source->body in the parsed query ($datum['parsedQuery']).

For now I decided against this approach: this requires lots of "internal detail knowledge" about the AST, which I'd rather not add as I don't see much overhead storing the original query (compared to the much bigger AST already added, too).

Reply:
Ok, I was also quite undecided about it.